### PR TITLE
editor: add confirmation prompt when executing clear editor history

### LINKED ISSUE #12496: editor: add confirmation prompt when executing `clear editor history`
#### Feature Description

The goal is to prompt end-users to confirm whether or not they want to actually clear their editor history (the history of recently opened editors). The confirmation will prevent clearing history accidentally for instance, and aligns with the behavior present in VS Code.

#### Related Code

_Command Registration_:

https://github.com/eclipse-theia/theia/blob/18b25035b8f0205a16c03ded32b6213f2cee7139/packages/editor/src/browser/editor-navigation-contribution.ts#L85-L88

_ConfirmDialog_:

Definition:

https://github.com/eclipse-theia/theia/blob/18b25035b8f0205a16c03ded32b6213f2cee7139/packages/core/src/browser/dialogs.ts#L360

Example Usage:

https://github.com/eclipse-theia/theia/blob/18b25035b8f0205a16c03ded32b6213f2cee7139/packages/core/src/browser/dialogs.ts#L394-L402

### PR DESCRIPTION
**Pull request by souravghosh01**
#### What it does
It adds a confirmation prompt when executing clear editor history 
fixes #12496

Recording

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/editor/src/browser/editor-navigation-contribution.ts [hunk 1]
```diff
@@ -83,7 +84,17 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
             isEnabled: () => !!this.locationStack.lastEditLocation()
         });
         this.commandRegistry.registerHandler(EditorCommands.CLEAR_EDITOR_HISTORY.id, {
-            execute: () => this.locationStack.clearHistory(),
+            execute: async () => {
+                const confirmed = await new ConfirmDialog({
+                    title: 'Clear Editor History',
+                    msg: 'Are you sure you want to clear the history of recently opened editors?',
```
Comment (vince-fugnitto) [inline packages/editor/src/browser/editor-navigation-contribution.ts:90]: We need to localize user-facing strings:

```suggestion
                    msg: nls.localizeByDefault('Do you want to clear the history of recently opened editors?'),
```

File: packages/editor/src/browser/editor-navigation-contribution.ts [hunk 2]
```diff
@@ -83,7 +84,17 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
             isEnabled: () => !!this.locationStack.lastEditLocation()
         });
         this.commandRegistry.registerHandler(EditorCommands.CLEAR_EDITOR_HISTORY.id, {
-            execute: () => this.locationStack.clearHistory(),
+            execute: async () => {
+                const confirmed = await new ConfirmDialog({
+                    title: 'Clear Editor History',
```
Comment (vince-fugnitto) [inline packages/editor/src/browser/editor-navigation-contribution.ts:89]:
```suggestion
                    title: nls.localizeByDefault('Clear Editor History'),
```

File: packages/editor/src/browser/editor-navigation-contribution.ts [hunk 3]
```diff
@@ -83,7 +84,17 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
             isEnabled: () => !!this.locationStack.lastEditLocation()
         });
         this.commandRegistry.registerHandler(EditorCommands.CLEAR_EDITOR_HISTORY.id, {
-            execute: () => this.locationStack.clearHistory(),
+            execute: async () => {
+                const confirmed = await new ConfirmDialog({
```
Comment (vince-fugnitto) [inline packages/editor/src/browser/editor-navigation-contribution.ts:88]: minor: we should probably rename in the context of the action

```suggestion
                const shouldClear = await new ConfirmDialog({
```

**Review by vince-fugnitto — COMMENTED** [review]
Thank you for your contribution :+1: 

In order to accept your changes please be sure to sure to sign the [eclipse contributor agreement (eca)](https://github.com/eclipse-theia/theia/blob/master/CONTRIBUTING.md#eclipse-contributor-agreement) with the same email as your authorship.

**Comment by souravghosh01** [comment]
@vince-fugnitto thank you so much for your suggestion. I have already signed the ECA

**Review by vince-fugnitto — APPROVED** [review]
LGTM :+1: Thank you for your first contribution!

**Pull request merged**